server/src/test/java/org/apache/druid/query/lookup/LookupReferencesManagerTest.java

```diff
@@ -526,6 +530,7 @@ public void testRealModeWithMainThread() throws Exception
     LookupExtractorFactory lookupExtractorFactory = EasyMock.createMock(LookupExtractorFactory.class);
     EasyMock.expect(lookupExtractorFactory.start()).andReturn(true).once();
     EasyMock.expect(lookupExtractorFactory.destroy()).andReturn(true).once();
+    EasyMock.expect(lookupExtractorFactory.isCacheLoaded()).andReturn(true).anyTimes();
```

Comment:
it would be nice to add some tests that simulate sad paths where the new lookup does not load to ensure proper behavior where the new container is destroyed and old container is still operational, and also tests where new doesn't initially load and so goes through the retry path but then successfully loads and the old is destroyed